include/functions.h

```diff
@@ -1889,76 +1888,57 @@ void UCodeDisas_Disassemble(UCodeDisas*, Gfx*);
-void func_800DE4B0(s32 poolIdx);
-void Audio_InitMainPools(s32 sizeForAudioInitPool);
-void Audio_SessionPoolsInit(AudioPoolSplit4* split);
-void Audio_SeqAndBankPoolInit(AudioPoolSplit2* split);
-void Audio_PersistentPoolsInit(AudioPoolSplit3* split);
-void Audio_TemporaryPoolsInit(AudioPoolSplit3* split);
-void* Audio_AllocBankOrSeq(s32 poolIdx, s32 size, s32 arg2, s32 id);
-void* func_800DF074(s32 poolIdx, s32 arg1, s32 id);
-void* func_800DF0CC(s32 poolIdx, s32 arg1, s32 id);
-void func_800DF5DC(s16* arg0, s32 arg1);
-void func_800DF630(s16* arg0, s32 arg1);
-void func_800DF688(s16* arg0, s32 arg1, s32 arg2);
-void func_800DF7C4(void);
-void func_800DF888(void);
-s32 Audio_ResetStep(void);
-void Audio_InitHeap(void);
-void* func_800E04E8(s32 poolIdx, s32 id);
-void* func_800E0540(s32 poolIdx, s32 id, u32 size);
-void* func_800E05C4(u32 size, s32 arg1, void* arg2, s8 arg3, s32 arg4);
-void func_800E0634(u32 arg0, u32 arg1);
-UnkHeapEntry* func_800E06CC(u32 size);
-void func_800E0964(UnkHeapEntry* entry, s32 bankId);
-void func_800E0AD8(UnkHeapEntry* entry);
-void func_800E0BB4(UnkHeapEntry* entry, AudioBankSample* sample);
-UnkHeapEntry* func_800E0BF8(u32 size);
-// ? func_800E0C80(?);
-void func_800E0CBC(void);
-void func_800E0E6C(s32 id);
-void func_800E0E90(s32 id);
-void func_800E0EB4(s32 arg0, s32 id);
-void func_800E1148(void);
-void func_800E11F0(void);
-void* Audio_DmaSampleData(u32 devAddr, u32 size, s32 arg2, u8* dmaIndexRef, s32 medium);
-void func_800E1618(s32 arg0);
-s32 Audio_IsBankLoadComplete(s32 bankId);
-s32 Audio_IsSeqLoadComplete(s32 seqId);
-void Audio_SetBankLoadStatus(s32 bankId, s32 status);
-void Audio_SetSeqLoadStatus(s32 seqId, s32 status);
-void func_800E1C18(s32 channelIdx, s32 arg1);
-s32 func_800E1D64(s32 arg0, s32 arg1, s32 arg2);
-void Audio_AudioSeqAsyncLoad(s32 arg0, s32 arg1, s32 arg2, OSMesgQueue* arg3);
-void Audio_AudioTableAsyncLoad(s32 arg0, s32 arg1, s32 arg2, OSMesgQueue* arg3);
-void Audio_AudioBankAsyncLoad(s32 arg0, s32 arg1, s32 arg2, OSMesgQueue* arg3);
-u8* func_800E1F38(s32 arg0, u32* arg1);
-void func_800E1F7C(s32 arg0);
-s32 func_800E20D4(s32 playerIndex, s32 seqId, s32 arg2);
-s32 func_800E2124(s32 playerIndex, s32 seqId, s32 arg2);
-void Audio_ProcessLoads(s32 resetStatus);
-void func_800E301C(void* callback);
-void Audio_ContextInit(void* heap, u32 heapSize);
-void Audio_SyncLoadsInit(void);
-s32 Audio_SyncLoadSample(s32 arg0, s32 arg1, s8* arg2);
-s32 Audio_SyncLoadSeq(s32 seqIdx, u8* ramAddr, s8* arg2);
-void Audio_AsyncLoadReqInit(void);
-void func_800E4D94(void);
-void func_800E4EEC(s32 tableType, s32 arg1, s8* arg2);
-void func_800E4F58(void);
-void func_800E4FB0(void);
+void AudioHeap_DiscardFont(s32 fontId);
+void AudioHeap_DiscardSequence(s32 seqId);
+void AudioHeap_WritebackDCache(void* mem, u32 size);
+void* AudioHeap_AllocZeroedMaybeExternal(AudioAllocPool* pool, u32 size);
+void* AudioHeap_AllocMaybeExternal(AudioAllocPool* pool, u32 size);
```

Comment:
are all of these functions expected to be used outside of the system? we can start removing "local" functions from this file if we feel its appropriate. i know nothing about these files to make that call myself

Reply:
They are used outside of the file they are defined in (or like AudioHeap_AllocZeroedMaybeExternal aren't but make sense to extern for symmetry with AudioHeap_AllocMaybeExternal), but not outside of the audio subsystem. Same goes for a majority of audio functions in functions.h. At some point I agree we should make a header for audio internals and move the declarations there, but not in this PR.